File: 1189-maximum-number-of-balloons/1189-maximum-number-of-balloons.cpp
class Solution {
public:
    int maxNumberOfBalloons(string text) {
         int b=0,a=0,l=0,o=0,n=0;
        
        for(int i=0;i<text.size();i++){
            if(text[i]=='b')b++;
            if(text[i]=='a')a++;
            if(text[i]=='l')l++;
            if(text[i]=='o')o++;
            if(text[i]=='n')n++;
        }
        l/=2;
        o/=2;
         return min(b,min(a,min(l,min(o,n))));
    }
};

File: 1189-maximum-number-of-balloons/1189-maximum-number-of-balloons.java
class Solution {
    public int maxNumberOfBalloons(String text) {
       
        int [] freq = new int[26]; //This array is used to count the frequency of the array 
        for(char ch : text.toCharArray()){
            freq[ch - 'a']++; //Frequency ++
        }
        
        char [] balloon = new char[]{'b' , 'a' , 'l' , 'o' , 'n'};
        int minfreq = Integer.MAX_VALUE;
        for(char c : balloon){
            if(c == 'l' || c == 'o'){
            freq[c-'a'] = freq[c-'a']/2;
            }
        
        minfreq = Math.min(minfreq , freq[c-'a']);
    }
    return minfreq;
        
    }
}